### PR TITLE
test: add new test TestPrebuildAndRegularWorkspaceDifferentWorkspaceClass

### PR DESCRIPTION
**Pull request by jenting**
## Description

Add a new test to test prebuild and regular workspace with different workspace class
- Prebuild small workspace class (20Gi) + regular large workspace class (30Gi): the workspace can start
- Prebuild large workspace class (30Gi) + regular small workspace class (20Gi): the workspace can't start because the restore volume snapshot size > the PVC size

## Related Issue(s)

Partially #12638

## How to test

```console
cd test

go test  -run ^TestPrebuildAndRegularWorkspaceDifferentWorkspaceClass$ github.com/gitpod-io/gitpod/test/tests/components/ws-manager -count=1 -namespace=default -kubeconfig=/home/gitpod/.kube/config -v
```

## Release Notes

```release-note
[experimental] show an error if the PVC size is less than the restore volume snapshot
```

## Documentation

None

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-integration-tests=workspace
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by jenting** [comment]
/hold
wait for CI pass

**Comment by utam0k** [comment]
This PR includes testing, but isn't the main focus of this PR a change to ws-manager? Since I think this PR provides value to users, why don't you write a Release Note?

File: test/tests/components/ws-manager/prebuild_test.go [hunk 1]
```diff
@@ -491,6 +408,263 @@ func TestOpenWorkspaceFromPrebuild(t *testing.T) {
...
+					ContextURL:       "https://github.com/gitpod-io/empty",
+					CheckoutLocation: "empty",
+					WorkspaceRoot:    "/workspace/empty",
+				},
+				{
+					Name: "prebuild-large-regular-small-workspace-class",
+					// TODO: do not use hard-code workspace class name to prevent if we change to different environment to run the test
+					PrebuildWorkspaceClass: "default",
+					RegularWorkspaceClass:  "small",
+					//
```
Comment (utam0k) [inline test/tests/components/ws-manager/prebuild_test.go:451]:
```suggestion
```

Comment (jenting) [reply test/tests/components/ws-manager/prebuild_test.go:451]: I intended to add this line to know which parts are the TODO.

**Comment by jenting** [comment]
> This PR includes testing, but isn't the main focus of this PR a change to ws-manager? Since I think this PR provides value to users, why don't you write a Release Note?

Release note added.

**Review by utam0k — APPROVED** [review]

**Comment by jenting** [comment]
> /hold

/unhold

**Pull request merged**